imap/monitor.go

```diff
@@ -147,41 +150,46 @@ func checkForNewEmails(im models.IMAP) {
 				}
 			}
 
-			body := string(append(m.Email.Text, m.Email.HTML...)) // Not sure if we need to check the Text as well as the HTML. Perhaps sometimes Text only emails won't have an HTML component?
-			rid := goPhishRegex.FindString(body)
+			rids, err := checkRIDs(m.Email) // Search email Text, HTML, and each attachment for rid parameters
 
-			if rid != "" {
-				rid = rid[5:]
-				log.Infof("User '%s' reported email with rid %s", m.Email.From, rid)
-				result, err := models.GetResult(rid)
-				if err != nil {
-					log.Error("Error reporting GoPhish email with rid ", rid, ": ", err.Error())
-					reportingFailed = append(reportingFailed, m.UID)
-				} else {
-					err = result.HandleEmailReport(models.EventDetails{})
+			if err != nil {
+				log.Errorf("Error searching email for rids from user '%s': %s", m.Email.From, err.Error())
+			} else {
```

Comment:
What do you think about removing this `else` and moving this back up a level?

Reply:
If I understand correctly, remove the `else` and change the error check to have a continue to step over to the next email?

```
if err != nil {
	log.Errorf("Error searching email for rids from user '%s': %s", m.Email.From, err.Error())
	continue
} 
if len(rids) < 1 { .......
```

Reply:
Right. It looks to me that this would fire if something broke when checking for the regex in the email content.

This would mean that we can't trust the result of `rids`. At that point, since the rest of the loop isn't applicable, we should just continue on with the next email.